api/src/test/java/org/openmrs/OrderTest.java

```diff
@@ -197,11 +201,22 @@ public void copy_shouldCopyAllFields() throws Exception {
 	 */
 	@Test
 	public void cloneForRevision_shouldSetAllTheRelevantFields() throws Exception {
-		assertThatAllFieldsAreCopied(new Order(), "cloneForRevision", "creator", "dateCreated", "action", "changedBy",
-		    "dateChanged", "voided", "dateVoided", "voidedBy", "voidReason", "encounter", "orderNumber", "orderer",
-		    "previousOrder", "dateActivated", "dateStopped", "accessionNumber");
+		Order newOrder = new Order();
+
+		OrderGroup orderGroup = new OrderGroup();
+		newOrder.setOrderGroup(orderGroup);
+
+		Order revisedOrder = newOrder.cloneForRevision();
+
+		assertThatAllFieldsAreCopied(revisedOrder, "cloneForRevision", "creator", "dateCreated", "action", "changedBy",
+				"dateChanged", "voided", "dateVoided", "voidedBy", "voidReason", "encounter", "orderNumber", "orderer",
+				"previousOrder", "dateActivated", "dateStopped", "accessionNumber", "orderGroup", "sortWeight");
+
+		assertNull("Revised order should not have orderGroup", revisedOrder.getOrderGroup());
```

Comment:
A revised order is just like any other order so it can have an order group and sortWeight